hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/io/TestHoodieTimelineArchiveLog.java

```diff
@@ -183,6 +217,41 @@ public void testArchiveTableWithArchival(boolean enableMetadata) throws Exceptio
     }
   }
 
+  @ParameterizedTest
+  @MethodSource("testArchiveTableWithArchivalTrim")
+  public void testArchiveTableWithArchivalCleanUp(boolean enableMetadata, boolean enableArchiveTrim, int archiveFilesToKeep) throws Exception {
+    HashSet<String> archiveFilesExisted = new HashSet<>();
+    ArrayList<String> currentExistArchiveFiles = new ArrayList<>();
+    HoodieWriteConfig writeConfig = initTestTableAndGetWriteConfig(enableMetadata, 2, 3, 2, enableArchiveTrim, archiveFilesToKeep);
+    String archivePath = metaClient.getArchivePath();
+    for (int i = 1; i < 10; i++) {
+      testTable.doWriteOperation("0000000" + i, WriteOperationType.UPSERT, i == 1 ? Arrays.asList("p1", "p2") : Collections.emptyList(), Arrays.asList("p1", "p2"), 2);
+      // trigger archival
+      archiveAndGetCommitsList(writeConfig);
+      RemoteIterator<LocatedFileStatus> iter = metaClient.getFs().listFiles(new Path(archivePath), false);
+      ArrayList<String> files = new ArrayList<>();
+      while (iter.hasNext()) {
+        files.add(iter.next().getPath().toString());
+      }
+      archiveFilesExisted.addAll(files);
+      currentExistArchiveFiles = files;
+    }
+
+    assertEquals(archiveFilesToKeep, currentExistArchiveFiles.size());
+
+    if (enableArchiveTrim) {
+      // sort archive files path
+      List<String> sorted = archiveFilesExisted.stream().sorted().collect(Collectors.toList());
+      List<String> archiveFilesDeleted = sorted.subList(0, 3 - archiveFilesToKeep);
+      List<String> archiveFilesKept = sorted.subList(3 - archiveFilesToKeep, sorted.size());
+
+      // assert older archive files are deleted
+      assertFalse(currentExistArchiveFiles.containsAll(archiveFilesDeleted));
+      // assert most recent archive files are preserved
+      assertTrue(currentExistArchiveFiles.containsAll(archiveFilesKept));
+    }
```

Comment:
Add a check when archive trim is disabled as well?

Reply:
Added.